tsdb/index/tsi1/cache.go

```diff
@@ -158,16 +165,21 @@ func (c *TagValueSeriesIDCache) delete(name, key, value []byte, x uint64) {
 			if ele, ok := tkmap[string(value)]; ok {
 				if ss := ele.Value.(*seriesIDCacheElement).SeriesIDSet; ss != nil {
 					ele.Value.(*seriesIDCacheElement).SeriesIDSet.Remove(x)
+					c.tracker.IncDeletesHit()
+					return
 				}
 			}
 		}
 	}
+	c.tracker.IncDeletesMiss()
 }
 
 // checkEviction checks if the cache is too big, and evicts the least recently used
 // item if it is.
 func (c *TagValueSeriesIDCache) checkEviction() {
-	if c.evictor.Len() <= c.capacity {
+	l := c.evictor.Len()
+	c.tracker.SetSize(uint64(l))
```

Comment:
This feels like buggy logic to me...

`checkEviction()` is only called at the end of `Put`, so we're not decrementing the size on `Delete`. We're also not readjusting the size after we evict items.